CMakeLists.txt

```diff
@@ -41,61 +31,91 @@ find_package(rclpy REQUIRED)
 find_package(ros2_shared REQUIRED)
 find_package(sensor_msgs REQUIRED)
 
-include_directories(include)
-
 # Create ament index resource which references the libraries in the binary dir
 set(node_plugins "")
 
+# How to build a library:
+# https://docs.ros.org/en/humble/How-To-Guides/Ament-CMake-Documentation.html#building-a-library
+
 #=============
 # GSCam node
 #=============
 
-set(GSCAM_NODE_DEPS
-  camera_calibration_parsers
-  camera_info_manager
-  class_loader
-  rclcpp
-  rclcpp_components
-  ros2_shared
-  sensor_msgs)
-
-add_library(gscam_node SHARED)
-target_sources(gscam_node PRIVATE src/gscam_node.cpp)
+add_library(gscam_node SHARED src/gscam_node.cpp)
```

Comment:
I recommend removing the SHARED keyword unless you want to force users to be unable to use static libs. See https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html

This would apply to any `add_library` call in the repo.

Reply:
Interesting, I'll do that.